# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: dom/workers/WorkerPrivate.cpp
# Commit: e4dbb1e6c722
# Full Hash: e4dbb1e6c72201d69477e5686de0cead6e063ca1
# Author: Eden Chuang <echuang@mozilla.com>
# Date: 2023-07-11 21:36:59
# Description:
#   Bug 1837856 - Notify WeakWorkerRefs in WorkerPrivate::RunLoopNeverRan. r=asuth
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D183077
# ==============================================================================

diff -r 4a5dbbfe3bf3 -r e4dbb1e6c722 dom/workers/WorkerPrivate.cpp
--- a/dom/workers/WorkerPrivate.cpp	Tue Jul 11 12:23:23 2023 +0300
+++ b/dom/workers/WorkerPrivate.cpp	Tue Jul 11 09:25:44 2023 +0000
@@ -3142,8 +3142,11 @@
     }
   }
 
-  // Should not have any WorkerRefs, children workers and Timeouts.
-  MOZ_DIAGNOSTIC_ASSERT(!HasActiveWorkerRefs());
+  // There should be no StrongWorkerRefs, child Workers, and Timeouts, but
+  // WeakWorkerRefs could. WorkerThreadPrimaryRunnable could have created a
+  // PerformanceStorageWorker which holds a WeakWorkerRef.
+  // Notify WeakWorkerRefs with Dead status.
+  NotifyWorkerRefs(Dead);
 
   ScheduleDeletion(WorkerPrivate::WorkerRan);
 }
